File: proyecto/src/app/model/cuenta.ts
import { Servicio } from './servicio';

export interface Cuenta {
  id?: number;
  estado?: string;  // ABIERTA, CERRADA, PAGADA, PENDIENTE
  total?: number;
  reserva?: { 
    id: number; 
    fechaInicio: string; 
    fechaFin: string; 
    estado: string; 
  };
  servicios?: Servicio[];
}

File: demo/src/main/java/com/moravia/demo/model/Cuenta.java
package com.moravia.demo.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonBackReference;

@Data
@Entity
@NoArgsConstructor
@Builder
@AllArgsConstructor
public class Cuenta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String estado;
    private Double total;

    @OneToOne
    @JoinColumn(name = "id_reserva")
    @JsonBackReference("reserva-cuenta")
    private Reserva reserva;

    @ManyToMany
    @JoinTable(
        name = "cuenta_servicios",
        joinColumns = @JoinColumn(name = "cuenta_id"),
        inverseJoinColumns = @JoinColumn(name = "servicio_id")
    )
    private List<Servicio> servicios;
}
